### DISCUSSION TITLE
Default query params

### DISCUSSION THREAD

**Discussion by stychu** [ecyrbe/zodios#225] [Q&A] [answered]
I was expecting something like this default will work. But seems it doesn't append the query param automatically??

```
const searchLocations = makeEndpoint({
  method: "get",
  path: `/v4/locations`,
  alias: "searchLocations",
  description: "Search for given locations",
  response: locationsSchema,
  parameters: [
    {
      name: "search",
      type: "Query",
      schema: z.string().nullish(),
    },
    {
      name: "filter:type",
      type: "Query",
      schema: z.string().default(LOCATION_BASIC_TYPES),
    },
  ],
});
```
```
[Request] 17:43:35 GET http://localhost.../v4/locations {"search":"abi"}
```
I expected it will will put the `filter:type` with the default

**Comment by ecyrbe** [comment, marked as answer]
Hello,

Yes, this is by design. if your backend does apply default values, no need for your frontend to send it.
For example, if you use `@zodios/express`, default parameter will be set on backend with this schema.

**Reply by stychu** [reply]
So if I use only zodios Client I can't pretty much do any preprocessing/transformation on query params??

**Reply by ecyrbe** [reply]
No, preprocessing and transformations will work on frontend, only default parameters will not be generated

**Reply by stychu** [reply]
But won't work if I use it on backend? Im using zodios Client to make Http calls to external API from my webapp backend (nextjs API)

**Reply by ecyrbe** [reply]
Here is what happen, transformations are done on frontend and disabled on backend to not transform twice. Default values are generated on backend.
You can enable transforms on backend and disable them on frontend (there is an option to control this behaviour) if you want it that way.
On client you just need to pass zodios contructor the option `transform: false` and on backend `transform: true`.

**Reply by ecyrbe** [reply]
Here are the docs : 
- [frontend](http://www.zodios.org/docs/client#options)
- [backend](http://www.zodios.org/docs/server/express-app#options)

**Reply by stychu** [reply]
Hmm I don't seem to have `transform` option 🤔 

![Screenshot 2022-11-07 at 19 36 52](https://user-images.githubusercontent.com/9383657/200388539-029104f6-62d4-49e6-ae93-7699fed2e4ad.png)

**Reply by stychu** [reply]
Ok nvm I've upgraded to latest and its there. But for some reason `transform` on query param doesn't work when value is empty string.

There is no call of transform when empty string is passed

![Screenshot 2022-11-07 at 19 49 50](https://user-images.githubusercontent.com/9383657/200390914-e7fee781-1eab-44f2-8b0b-6ce6f2298c69.png)

**Reply by ecyrbe** [reply]
This is definitively a bug.  i'll create an issue and a bugfix for it.

**Reply by ecyrbe** [reply]
This is now fixed on on `@zodios/core` v10.6.2

**Reply by ecyrbe** [reply]
a new option to activate sending default values is available on `@zodios/core` v10.7.0
cf : http://www.zodios.org/docs/client#options

**Reply by stychu** [reply]
@ecyrbe This is brilliant! Thank you. All working as expected now 💪 💪 💪
